oxidation/crates/parsec_api_protocol/src/error.rs

```diff
@@ -0,0 +1,49 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BSLv1.1 (eventually AGPLv3) 2016-2021 Scille SAS
+
+use parsec_api_types::DateTime;
+use thiserror::Error;
+
+use crate::ApiVersion;
+
+#[derive(Error, Debug, PartialEq)]
+pub enum HandshakeError {
+    #[error("{0}")]
+    FailedChallenge(String),
+    #[error("Handshake bad administration token {0:?}")]
+    BadAdministrationToken(Option<String>),
```

Comment:
Regarding all those `Option<String>` fields, I think we should remove them altogether: they are based on the `help` field from handshake which is similar to the `reason` field in the command response. In both case those fields are too broad to be used in a programmatic way (no guarantee what the field value would be, and parsing super cumbersome if it contains variable such as user id) and too low-level to be displayed to the user.

Reply:
Done :+1: